lua/starfall/sflib.lua

```diff
@@ -634,12 +634,15 @@ function SF.Require(moduleName)
 	return false
 end
 
-
-function SF.CompileString(str, name, handle)
-	if string.find(str, "repeat.*continue.*until") then
-		return "Due to a glua bug. Use of the string 'continue' in repeat-until loops has been banned"
+--- Compile String but fix a compile error.
+function SF.CompileString(script, identifier, handle_error)
+	if
+		script:match("[^%w%d_]repeat[^%w%d_].*[^%w%d_]continue[^%w%d_].*[^%w%d_]until[^%w%d_]")
+		or script:match("[^%w%d_]repeat[^%w%d_]+continue[^%w%d_]+until[^%w%d_]")
```

Comment:
Sorry, I nearly forgot about this, but what case is this second match covering that the first one doesn't?

Reply:
When there's only one character between the keywords, I put it in the commit message, should've put it in a comment. Don't know how else to do it with a whitelist match

```lua
local dat = [[
repeat continue until(true)
]]

if
	dat:match("[^%w%d_]repeat[^%w%d_].*[^%w%d_]continue[^%w%d_].*[^%w%d_]until[^%w%d_]")
	--or dat:match("[^%w%d_]repeat[^%w%d_]+continue[^%w%d_]+until[^%w%d_]")
then
	print "Matched"
else
	print "Nope"
end

--> Nope
```